### PR TITLE
fix(ecma/lexer): hexadecimal escape sequences

### LINKED ISSUE #2822: TaggedTemplateLiteral Parser Error
### Describe the bug

When dealing the template object, there may be an error in swc,
the result of input code run by node is true, but after transforming by swc, the result is false.

### Input code

```typescript
let filter = void 0;
function test(quasis, ...expressionValues){
	const {raw} = quasis;
	const rawLen = raw.length;
	expressionValues = expressionValues.map( filter || ((x) => `(${x})`) )

	let s = '', i = 0;
	while (true) {
		s += raw[i];
		if (i + 1 === rawLen) {
			return s;
		}
		s += expressionValues[i++];
	}
}
{// raw unicode, hex
	let hex = test`\x22\x21\x224`;
	console.log(hex === "\\x22\\x21\\x224");
}
```

### Config

```json
{
  "jsc": {
    "parser": {
      "syntax": "ecmascript",
      "jsx": false,
      "dynamicImport": false,
      "privateMethod": false,
      "functionBind": false,
      "exportDefaultFrom": false,
      "exportNamespaceFrom": false,
      "decorators": false,
      "decoratorsBeforeExport": false,
      "topLevelAwait": false,
      "importMeta": false
    },
    "transform": null,
    "target": "es5",
    "loose": false,
    "externalHelpers": false
  }
}
```

### Playground link

https://play.swc.rs/?version=1.2.111&code=H4sIAAAAAAAAA12QTWrDMBCF19IphlCIhI3ThOyMe4Kuu0kKNs4kFqhSqp%2FG4PjuHdnxxgsxw3tvRh%2BjMcBV6YAOKviz6gLvJb9G0wZlDQT0QfzGxiufQ1EU2N8dek%2FWV6Mjejlw1lrjAwyueYy0Yg6Xi0zqJxrSqSk0mlvoyFuvIX8tFT%2FNXSxkzycI0UuoPqAWb0M%2FylqC5Jxpok%2FT220Oiiqxs0enNIIILqIE4mMesun%2Fk%2Fomm6krCAUZ7KGqqhfgHGQOQ3QGEj4bl8k12UllWVo08pEPu13aANGo1l4whw77mYoa4kn3q8%2F94UBvn%2Bqxfp3Gaiy0vYkpRxyb8xSbclNwI0s%2B%2FgNMedErngEAAA%3D%3D&config=H4sIAAAAAAAAA0VOOw6DMAy9i2fWLtygA4ewgkFBSRzZRmqKuHsTNcBkv4%2Ff8wGbOhgPyChK0jYtyfADI5CLqE58NhiqrVILBqVzAENZyZpFX1ULzEpdHSD65JfSkhzHLKT6SJjWcDnPGhR53htxgJVcZzuJnDatqWrind23fzjx%2FDQF%2FJYbJH4nI%2BH8vHl90h1ep15nstP5A1SlDpf7AAAA

### Expected behavior

The ori code run by node is true, but after transforming, the result is false.

### Version

1.2.111

### Additional context

I run swc use cmd as follows
```bash
npx swc --config-file=.swcrc poc.js -o poc.js
```
The environment is ubuntu 20.04 amd64.

### PR DESCRIPTION
**Pull request by RiESAEX**
fix #2822

### CODE REVIEW

**Review by kdy1 — APPROVED** [review]
Thanks!

**Comment by kdy1** [comment]
Oh, some tests are failing.
Can you run `UPDATE=1 cargo test` from parser and minifier?

**Comment by RiESAEX** [comment]
sure, but I think transform 
```js
`\x3c!--` to "\\x3c--"
```
 isn't right for a minifer. I'm looking into it, any idea?

File: crates/swc_ecma_minifier/src/compress/pure/strings.rs [hunk 1]
```diff
@@ -23,6 +23,7 @@ where
                 }) && (M::force_str_for_tpl()
                     || (!c.value.contains("\\n") && !c.value.contains("\\r")))
                     && !c.value.contains("\\0")
+                    && !c.value.contains("\\x")
```
Comment (RiESAEX) [inline crates/swc_ecma_minifier/src/compress/pure/strings.rs:26]: before this, we store 
```js
`\x21`
```
like \0x21, hit line 25.
now it is \x21 so I add this line

**Review by kdy1 — APPROVED** [review]
Thanks!

**Pull request merged**